js/src/logger.ts

```diff
@@ -1642,6 +1655,87 @@ function validateAndSanitizeExperimentLogFullArgs(
   return event;
 }
 
+class ObjectFetcher<RecordType> {
+  private excludeFields: Set<string> = new Set(["_object_delete"]);
+  private _fetchedData: any[] | undefined = undefined;
+
+  constructor(
+    private objectType: "dataset" | "experiment",
+    private jsonFields: Set<string>,
+    private pinnedVersion: string | undefined
+  ) {}
+
+  public get id(): Promise<string> {
+    throw new Error("ObjectFetcher subclasses must have an 'id' attribute");
+  }
+
+  protected async getState(): Promise<BraintrustState> {
+    throw new Error("ObjectFetcher subclasses must have a 'getState' method");
+  }
+
+  async *fetch(): AsyncGenerator<RecordType> {
+    const records = await this.fetchedData();
+    for (const record of records) {
+      yield Object.fromEntries(
+        Object.entries(record)
+          .filter(([k, _]) => !this.excludeFields.has(k))
+          .map(([k, v]) => [
+            k,
+            this.jsonFields.has(k) && typeof v === "string" ? JSON.parse(v) : v,
+          ])
+      ) as RecordType;
+    }
+    this.clearCache();
```

Comment:
Why does using the fetch iterator trigger a cache clear?

Reply:
Hmm, I copied this over from the last thing, but it's likely an incorrect relic of debugging code. Will remove.